fe/fe-core/src/main/java/org/apache/doris/analysis/QueryStmt.java

```diff
@@ -301,7 +301,7 @@ protected void createSortInfo(Analyzer analyzer) throws AnalysisException {
         sortInfo = new SortInfo(orderingExprs, isAscOrder, nullsFirstParams);
         // order by w/o limit and offset in inline views, set operands and insert statements
         // are ignored.
-        if (!hasLimit() && !hasOffset() && !analyzer.isRootAnalyzer()) {
+        if (!hasLimit() && !hasOffset() && (!analyzer.isRootAnalyzer() || fromInsert)) {
```

Comment:
May be we could just remove the order by directly. 
Because the value of Insert is actually useless after order by.

Reply:
I read it wrong, it's okay